qemu/lib/iozone.py

```diff
@@ -0,0 +1,164 @@
+    """
+    def __init__(self, vm, params, test):
+        """
+        run setup of iozone test
+        """
+        self.params = params
+        self.vm = vm
+        self.session = vm.wait_for_login()
+        self.test = test
+        self.results_path = ""
+        self.timeout = float(params.get("iozone_timeout", 240))
+
+    def before_run_iozone(self):
+        """
+        Prepare for iozone test
+        """
+        pass
+
+    def run_iozone_in_guest(self, disk_index, mountpoint):
+        """
+        Run iozone test in guest
+
+        :param disk_index: index to find disk under test
+        :param mountpoint: mountpoint/disk letter assigned to disk under test
+        :return: iozone run status and result.
+        """
+        iozone_test_cmd_default = "./iozone -azR -r 64k -n 125M -g 512M -i 0 -i 1 %s"
+        iozone_test_cmd = self.params.get("iozone_test_cmd", iozone_test_cmd_default)
+        if self.params.get("os_type") == "windows":
+            if self.params.get("format_disk", "yes") == "yes":
+                logging.info("Format disk")
+                utils_misc.format_windows_disk(self.session, disk_index, mountpoint)
+            test_cmd = iozone_test_cmd % (mountpoint, mountpoint)
+            test_cmd = utils_misc.set_winutils_letter(self.session, test_cmd)
+        elif self.params.get("os_type") == "linux":
+            test_cmd = iozone_test_cmd % mountpoint
+            self.session.cmd_output_safe('cd %s' % self.iozone_work_path)
+
+        status, output = self.session.cmd_status_output(test_cmd,
+                                                        timeout=self.timeout)
+        self.results_path = os.path.join(self.test.resultsdir, 'raw_output_%s'
+                                         % self.test.iteration)
+        logging.info("Write results to %s" % self.results_path)
+        with open(self.results_path, 'w') as file:
+            file.write(output)
+
+        return status, output
+
+    def after_run_iozone(self):
+        """
+        Clean up iozone env/post result after iozone test
+        """
+        pass
+
+
+class Iozone_Win(IozoneTest):
+    """
+    Check driver before iozone test, post result after iozone test.
+    """
+    def __init__(self, vm, params, test):
+        IozoneTest.__init__(self, vm, params, test)
+
+    def get_driver(self):
+        """
+        Get driver name according to driver_format
+        """
+        driver_name = self.params.get("driver_name", "")
+        drive_format = self.params.get("drive_format")
+        if not driver_name:
+            if "scsi" in drive_format:
+                driver_name = "vioscsi"
+            elif "virtio" in drive_format:
+                driver_name = "viostor"
+            else:
+                driver_name = None
+        return driver_name
+
+    def driver_verify(self):
+        """
+        Run driver verification if needed
+        """
+        driver_name = self.get_driver()
+        if driver_name:
```

Comment:
if `driver_name` is `None`, it should raise error here, or 
when `get_driver` returns `None` it raise error.